pkg/lib/oauth/handler/handler_token.go

```diff
@@ -1934,8 +1934,14 @@ func (h *TokenHandler) handleClientCredentials(
 	if r.Resource() == "" {
 		return nil, protocol.NewError("invalid_request", "resource is required")
 	}
+	if strings.HasPrefix(r.Resource(), h.IDTokenIssuer.Iss()) {
+		return nil, protocol.NewError("invalid_request", "invalid resource uri")
```

Comment:
In https://datatracker.ietf.org/doc/html/rfc8707#name-resource-parameter it introduces `invalid_target`. We should use that.

- When resource is empty, `invalid_target`, `resource is required`
- When resource matches our prefix, `invalid_target`, `resource URI must not be a prefix of iss`
- When resource is not found, `invalid_target`, `resource not found`
- When resource is not associated with the client, `invalid_target`, `client is not associated with the resource`

Reply:
For the second case, lets use `resource URI must not be a prefixed by authgear endpoint` which is clearer. `iss` in this context could be difficult to understand.

Reply:
Maybe just say `resource URI is reserved by Authgear`